src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb

```diff
@@ -1133,7 +1133,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             For i = 0 To argCount - 1
                 Dim arg As Integer = attrData.GetConstructorArgument(Of Integer)(i, SpecialType.System_Int32)
                 If arg < 0 Then
-                    diagnostics.Add(ERRID.ERR_BadAttribute1, If(nodeOpt IsNot Nothing, nodeOpt.ArgumentList.Arguments(i).GetLocation(), NoLocation.Singleton), attrData.AttributeClass)
+                    Dim location As Location
+                    If nodeOpt IsNot Nothing Then
+                        If nodeOpt.ArgumentList IsNot Nothing AndAlso nodeOpt.ArgumentList.Arguments.Count > i Then
+                            location = nodeOpt.ArgumentList.Arguments(i).GetLocation()
+                        Else
+                            location = nodeOpt.GetLocation()
+                        End If
+                    Else
+                        location = NoLocation.Singleton
+                    End If
```

Comment:
Is this the same code used in other `Source*Symbol` methods? If so, consider extracting a helper method.